classes/query/Repository.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+/**
+ * @file classes/query/Repository.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Repository
+ *
+ * @see Query
+ *
+ * @brief Operations for retrieving and modifying Query objects.
+ */
+
+namespace PKP\query;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\notification\NotificationManager;
+use APP\submission\Submission;
+use Illuminate\Support\Facades\Mail;
+use PKP\db\DAORegistry;
+use PKP\mail\Mailable;
+use PKP\note\Note;
+use PKP\notification\Notification;
+use PKP\notification\NotificationSubscriptionSettingsDAO;
+use PKP\security\Role;
+use PKP\stageAssignment\StageAssignment;
+use PKP\user\User;
+
+class Repository
+{
+    /**
+     * Retrieve a count of all open queries totalled by stage
+     *
+     * @param ?int[] $participantIds Only include queries with these participants
+     *
+     * @return array<int,int> [int $stageId => int $count]
+     */
+    public function countOpenPerStage(int $submissionId, ?array $participantIds = null): array
+    {
+        $counts = Query::withAssoc(Application::ASSOC_TYPE_SUBMISSION, $submissionId)
+            ->when($participantIds !== null, function ($q) use ($participantIds) {
+                $q->withUserIds($participantIds);
+            })
+            ->withClosed(false)
+            ->selectRaw('stage_id, COUNT(stage_id) as count')
+            ->groupBy('stage_id')
+            ->get()
+            ->mapWithKeys(fn ($row, $key) => [$row->stage_id => $row->count])
+            ->toArray();
+
+        return collect(Application::get()->getApplicationStages())
+            ->mapWithKeys(fn ($stageId, $key) => [$stageId => $counts[$stageId] ?? 0])
+            ->toArray();
+    }
+
+    /**
+     * Sequentially renumber queries in their sequence order.
+     *
+     * @param int $assocType Application::ASSOC_TYPE_...
+     * @param int $assocId Assoc ID per assocType
+     */
+    public function resequence($assocType, $assocId): void
+    {
+        $result = Query::withAssoc($assocType, $assocId)
+            ->orderBy('seq')
+            ->get();
+
+        $result->each(function ($item, $key = 1) {
+            Query::find($item->queryId)
+                ->update(['seq' => $key]);
```

Comment:
`upsert` will not work here as it will require exact field name but we can do something like 
```php
$result->each(function (Query $item, int $key = 1) {
    $item->update(['seq' => $key]);
});
```

here calling `find` again need to retrieve the row and then update it which seems double work to me as we already have those when we call `get` as Database Collection . Or I am missing something ?